workflow/usage.rst

```diff
@@ -0,0 +1,159 @@
+.. index::
+    single: Workflow; Usage
+
+How to Use the Workflow
+=======================
+
+Using the workflow component will help you to keep your domain logic as
+configuration. Having domain logic in one place gives you a better overview
+and it is easier to maintain whenever the domain requirement changes since
+you do not have to edit all your controllers, twig templates and services.
+
+A workflow is a process or a lifecycle that your objects go through. Each
+step or stage in the process is called a *place*. You do also define *transitions*
+to that describes the action to get from one place to another.
+
+.. image:: /_images/components/workflow/states_transitions.png
+
+A set of places and transitions creates a **definition**. A workflow needs
+a ``Definition`` and a way to write the states to the objects (i.e. an
+instance of a :class:`Symfony\\Component\\Workflow\\MarkingStore\\MarkingStoreInterface`.)
+
+Consider the following example for a blog post. A post can have places:
+'draft', 'review', 'rejected', 'published'. You can define the workflow
+like this:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        framework:
+            workflows:
+                blog_publishing:
+                    type: 'workflow' # or 'state_machine'
+                    marking_store:
+                        type: 'property_accessor' # or 'scalar' or 'state_machine'
```

Comment:
BTW, the type of marking store can not be `state_machine` ([see here](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L247))